dwave/optimization/include/dwave-optimization/graph.hpp

```diff
@@ -374,4 +374,8 @@ class DecisionNode : public Decision, public virtual Node {
     bool removable() const override { return false; }
 };
 
+std::vector<const Array*> cast_to_array(std::span<const ArrayNode* const> node_ptrs);
```

Comment:
Could/should replace with
```c++
auto cast_to_array(std::span<const ArrayNode* const> ptrs) {
    return std::ranges::transform_view(ptrs, [](auto ptr) -> const Array* { return ptr; });
}
```
or
```c++
template<ranges::viewable_range R>
auto cast_to_array(R&& ptrs) {
    return std::ranges::transform_view(ptrs, [](auto ptr) -> const Array* { return ptr; });
}
```
we could use `require` and `std::range_value_t` if we're worried about using ranges of other types.

Or, IMO best of all, just use `std::ranges::transform_view(ptrs, [](auto ptr) -> const Array* { return ptr; })` inline anywhere we might want this behavior and not add a new function at all!

Reply:
It is nice to have a version of this for `initializer_list` though, which I don't think we can use `transform_view` on?

Reply:
Switched to just two constructors on `ValuesInfo` for this, one for `initializer_list` and one that takes `viewable_range` directly so these methods are no longer necessary.

Reply:
... and that doesn't work with GCC11 so had to add the `span` constructor back.